### ISSUE TITLE
mutiple different calls to session.run in training

### ISSUE THREAD

**Issue by jmugan** [tensorflow/tensor2tensor#233]
tensor2tensor is very cool. I'd like to make different calls to `session.run` (or something equivalent) to train a wake-sleep version, with one call for wake and another call for sleep. Is there a way to do that with tensor2tensor? GANs have a similar requirement. I see that `cycle_gan.py` has a reverse gradient; possibly to get around this issue?

I might be missing something obvious, but it looks like tensor2tensor uses `Experiment._call_train()`, so I'm not sure how to do it.

**Comment by lukaszkaiser** [comment]
It's currently not easy due to our use of Experiment and Estimator and multi-problem support. We'd love to make it easier and welcome suggestions (and even better, PRs). One way to work around it though that's quite easy for simple schedules is to do sth along these lines:
```
loss = tf.cond(tf.equals(0, tf.mod(global_step, 2)),
              loss_function_1, loss_function_2)
```
So you're running the graph for loss1, say a generator, in even steps (or every 5 steps or so) and the graph for loss2 (e.g., discriminator) in odd steps. I admit it doesn't look pretty, but it might be a reasonable workaround until we clean up stuff.

In any case, let us know what works or doesn't work for you, thanks!

**Comment by jmugan** [comment]
Cool! Thanks!
